### PR TITLE
FileOFF for TriangleMesh

### PR DESCRIPTION
**Pull request by kongsgard**
Added IO support for OFF file format.

---
This change is [Reviewable](https://reviewable.io/reviews/intel-isl/open3d/1035)

### CODE REVIEW

File: src/Open3D/IO/FileFormat/FileOff.cpp [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+    for (int vidx = 0; vidx < num_of_vertices; vidx++) {
+        fscanf(file_in, "%f %f %f", &vx, &vy, &vz);
+        mesh.vertices_[vidx] = Eigen::Vector3d(vx, vy, vz);
+        utility::AdvanceConsoleProgress();
+    }
+
+    int n, vidx1, vidx2, vidx3;
+    for (int tidx = 0; tidx < num_of_triangles; tidx++) {
+        fscanf(file_in, "%d %d %d %d", &n, &vidx1, &vidx2, &vidx3);
+        if (n != 3) {
```
Comment (kongsgard) [inline src/Open3D/IO/FileFormat/FileOff.cpp]: OFF files can specify polygonal faces with more vertices than 3.
We could support such files by converting the polygonal faces to triangular faces with the following:
```
// Load all vertex indices into array vidxs
for (int i = 0; i < n-3; i++) {
    mesh.triangles_.push_back(Eigen::Vector3i(vidxs[0], vidxs[n+1], vidxs[n+2]))
}
```
Would this be a better solution?

**Comment by kongsgard** [comment]
Thanks for the great feedback on this PR!

**Comment by griegler** [comment]
Thanks @kongsgard for your work on this PR. Let's see if @yxlao or @qianyizh have some comments.

**Comment by yxlao** [comment]
Thanks @kongsgard, and @griegler for the review. Looks great.

**Pull request merged**